### PR TITLE
HiCExplorer

### PR DESCRIPTION
**Pull request by joachimwolff**
This PR adds galaxy wrappers for HiCExplorer, a software to analyze Hi-C data.
https://github.com/maxplanck-ie/HiCExplorer

### CODE REVIEW

**Comment by bgruening** [comment]
I'm running out of ideas here. This test is passing locally for me on two different machines.
The failure is also due to a "warning" which should not fail in this way ... any help welcome.

**Comment by bgruening** [comment]
@joachimwolff I have found this: https://github.com/ContinuumIO/anaconda-issues/issues/6678

**Comment by bgruening** [comment]
So this code snippet might help:
```
import warnings
warnings.filterwarnings("ignore", message="numpy.dtype size changed")
warnings.filterwarnings("ignore", message="numpy.ndarray size changed")
```
A bioconda patch would be nice for this.

**Comment by bgruening** [comment]
For reference: The `warnings` module behaves differently depending on the environment. I assume that travis marks the VM somehow as test environment, which will turn warnings into errors. HiCExplorer now filters the warnings upstream. So tests passes here.
@joachimwolff I did some cosmetic changes, as the stdio tag is not needed in our case.

**Comment by joachimwolff** [comment]
Can imho be merged if there are no more open issues.

File: tools/hicexplorer/hicCorrelate.xml [hunk 1]
```diff
@@ -53,7 +58,7 @@
             </when>
         </conditional>
         <repeat name="chromosomes" title="List of chromosomes to be included in the correlation" min="0">
-            <param name="chromosome" type="text"/>
+            <param name="chromosome" type="text" optional="false"/>
```
Comment (bgruening) [inline tools/hicexplorer/hicCorrelate.xml]: I think what you want to use is: `<validator type="empty_field" />`

Comment (bgruening) [reply tools/hicexplorer/hicCorrelate.xml]: e.g. here:https://github.com/galaxyproject/tools-iuc/blob/fbc18d9128669e461e76ed13307ee88dd774afa5/tool_collections/snpsift/snpsift/snpSift_rmInfo.xml

File: tools/hicexplorer/hicCorrelate.xml [hunk 2]
```diff
@@ -18,14 +18,19 @@
...
 
-            #if $use_range.select_use_range == "yes_use_range"
+            #if $use_range.select_use_range == "yes_use_range":
                 --range $range_min:$range_max
             #end if
 
             #if $chromosomes:
-                --chromosomes #echo '" "'.join([ '"%s"' % $chrom for $chrom in $chromosomes ])#
+                
+                #set $chromosome_list =  '" "'.join([ '"%s"' % $chrom.chromosome for $chrom in $chromosomes ])#
```
Comment (bgruening) [inline tools/hicexplorer/hicCorrelate.xml]: `"' '".join([ "'%s'" % $chrom.chromosome for $chrom in $chromosomes ])#`

Maybe this is better.

**Comment by bgruening** [comment]
Cool thanks @joachimwolff!

**Pull request merged**